dnf5daemon-server/services/goal/goal.cpp

```diff
@@ -293,30 +275,35 @@ sdbus::MethodReply Goal::do_transaction(sdbus::MethodCall & call) {
     dnfdaemon::KeyValueMap options;
     call >> options;
 
-    auto * transaction = session.get_transaction();
+    bool offline = dnfdaemon::key_value_map_get<bool>(options, "offline", false);
 
-    download_packages(session, *transaction);
+    if (offline) {
+        session.store_transaction_offline();
+        // TODO(mblaha): signalize reboot?
```

Comment:
Could you add API to check whether there's a pending offline update, please? It can be used by gnome-shell, to check whether the offline update is ready, then it asks the user `[x] Install pending update` in the power off/restart dialog and, when the user unchecks the option, the prepared update is "cancelled" (another new API) and the machine powers off/reboots without applying the offline update.

Reply:
Thanks for suggestions! I'll add them. We actually have similar funcionality on CLI already - `dnf offline status` and `dnf offline clean`.

Reply:
Added.

Reply:
I tested the `do_transaction` part and it works as expected. Thank you.